### PR TITLE
MM-57876: Use opensearch only

### PR DESCRIPTION
**Pull request by agnivade**
https://mattermost.atlassian.net/browse/MM-57876

### CODE REVIEW

File: deployment/config.go [hunk 1]
```diff
@@ -354,8 +354,16 @@ func (c *Config) validateElasticSearchConfig() error {
...
+		return fmt.Errorf("Incorrect engine version: %s. Must start with %q", c.ElasticSearchSettings.Version, "OpenSearch")
+	}
+
+	if c.ElasticSearchSettings.SnapshotRepository == "" {
+		return fmt.Errorf("Empty SnapshotRepository. Must supply a value")
+	}
+
+	if c.ElasticSearchSettings.SnapshotName == "" {
+		return fmt.Errorf("Empty SnapshotName. Must supply a value")
 	}
```
Comment (agarciamontoro) [inline deployment/config.go:367]: Can we create a ticket to allow deploying Opensearch without a snapshot?

File: deployment/opensearch/opensearch.go [hunk 2]
```diff
@@ -65,14 +63,13 @@ type Repository struct {
 
 // ListRepositories returns the list of repositories registered in the server
 func (c *Client) ListRepositories() ([]Repository, error) {
-	req := esapi.SnapshotGetRepositoryRequest{}
-	repositoriesResponse := make(map[string]repositoryResponse)
-	if err := c.get(req, &repositoriesResponse); err != nil {
+	resp, err := c.client.Snapshot.Repository.Get(context.Background(), nil)
```
Comment (agarciamontoro) [inline deployment/opensearch/opensearch.go:66]: Oh, this is much better than before :joy:

File: deployment/terraform/create.go [hunk 3]
```diff
@@ -994,9 +994,6 @@ func (t *Terraform) updateAppConfig(siteURL string, sshc *ssh.Client, jobServerE
 	cfg.SqlSettings.MaxIdleConns = model.NewPointer(100)
 	cfg.SqlSettings.MaxOpenConns = model.NewPointer(100)
 	cfg.SqlSettings.Trace = model.NewPointer(false) // Can be enabled for specific tests, but defaulting to false to declutter logs
-	if t.output.HasElasticSearch() {
-		cfg.SqlSettings.DisableDatabaseSearch = model.NewPointer(true)
-	}
```
Comment (agarciamontoro) [inline deployment/terraform/create.go:999]: Thanks for this! Can we split it to another PR, thought? It feels like we may need to point to this change in the future, and it's quite hidden here.

File: go.mod [hunk 4]
```diff
@@ -33,6 +33,7 @@ require (
 	github.com/graph-gophers/graphql-go v1.5.1-0.20230110080634-edea822f558a
 	github.com/mattermost/mattermost/server/public v0.1.7-0.20240806035841-540febd866aa
 	github.com/mattermost/mattermost/server/v8 v8.0.0-20240726090344-5547504c1d68
+	github.com/opensearch-project/opensearch-go/v4 v4.1.0
```
Comment (agarciamontoro) [inline go.mod:36]: I see this one was added, but the `github.com/elastic/go-elasticsearch/v7` dependency was not removed. Is it still needed?

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Thank you for this! The new client is soooo much better, it really simplifies the code.

Some additional questions:
- Have we tested this? Do you want a test from my side?
- Do we need to update the current snapshot of the 100M posts database dump? Or can we use the same one?

File: go.mod (hunk 4 above)
Comment (agnivade) [reply go.mod:36]: Yeah `go mod tidy` didn't remove it. We still use some elasticsearch types for the requests I believe.

File: deployment/terraform/create.go (hunk 3 above)
Comment (agnivade) [reply deployment/terraform/create.go:999]: ok

**Comment by agnivade** [comment]
> Have we tested this? Do you want a test from my side?

I have tested this.

> Do we need to update the current snapshot of the 100M posts database dump? Or can we use the same one?

I have restored the existing ES snapshot we have, and the restore works fine. Although I didn't use it with the 100M one, I don't think it would be an issue. The main thing to check was the restore and it works fine.

File: deployment/config.go (hunk 1 above)
Comment (agnivade) [reply deployment/config.go:367]: https://mattermost.atlassian.net/browse/MM-60336

**Review by agarciamontoro — APPROVED** [review]
Thank you!

**Review by streamer45 — APPROVED** [review]
Looks good!

**Pull request merged**